p34/src/biguint.rs

```diff
@@ -0,0 +1,212 @@
+// Implement in lib.rs a simplified BigUint4096 type, i.e. unsigned 4096 bit integer
+// ● Under the hood use array of u64s
+// ● Browse the standard library docs for potentially useful methods:
+// https://doc.rust-lang.org/stable/std/primitive.u64.html
+// ● Implement basic arithmetic traits
+// ● Implement conversion methods from hex-encoded &str and to hex-
+// encoded String (be careful about endianness)
+// ● Add integration tests for the type
+
+/// BigUint64
+/// Note: the Least Significant Bit (LSB) is stored on position 0
+#[derive(Debug, PartialEq)]
+pub struct BigUintGeneric<const SIZE: usize> {
+    value: [u64; SIZE],
+}
+
+// Source: https://doc.rust-lang.org/reference/items/type-aliases.html
+pub type BigUint4096 = BigUintGeneric<64>;
+
+// Source: https://doc.rust-lang.org/reference/items/generics.html
+// // Used as a field of a struct.
+// struct Foo<const N: usize>([i32; N]);
+//
+// impl<const N: usize> Foo<N> {
+//     // Used as an associated constant.
+//     const CONST: usize = N * 4;
+// }
+
+/// Bad implementation of a BigUint4096
+/// Warning: There might be defects. Do not use for production, this is only a sandbox where to test the implementation.
+impl<const N: usize> BigUintGeneric<N> {
+    pub fn from(initial_value: [u64; N]) -> Self {
+        Self {
+            value: initial_value,
+        }
+    }
+
+    pub fn new() -> Self {
+        Self::from([0; N])
+    }
+    pub fn sum(&mut self, another: &Self) {
```

Comment:
You should implement the `sub::ops` trait instead of these inherent methods.

Reply:
I implemented `std::ops::{Add, Mul}`, left AddAssign for later